### ISSUE TITLE
False positive match for DriveDroid

### ISSUE THREAD

**Issue by Kufat** [devadvance/rootcloak#96]
The mechanism that rootcloak uses to ID apps incorrectly identifies DriveDroid as com.google.android.gms. E.g.:
```
07-29 00:39:51.327 23740 23740 I Xposed  : Loaded app: com.google.android.gms
root@hammerhead:/ # cat /proc/23740/cmdline ; echo " "
com.softwarebakery.drivedroid
```

This happens across reboots and clearing RootCloak's data and still occurs with the latest beta on the appstore.

I'm not an Android developer, but is this a case of DriveDroid's argv[0] having the wrong executable name for wahtever reason? If so, I'll email DriveDroid's developer...but it'd also be good if rootcloak could more reliably determine the real executable/package name.

(Would this present a mechanism for root-detecting apps to spoof the names of apps that root users always want to allow, such as Titanium Backup, or is there a special case that causes DriveDroid to appear as com.google.android.gms?)

**Comment by hikaritenchi** [comment]
I haven't been able to reproduce your results, unfortunately. When I open DriveDroid, I get 
`07-29 02:49:22.728 7475-7475/? I/Xposed: Loaded app: com.softwarebakery.drivedroid`

We use the package name as ultimately provided by Xposed, but presumably it's the same one that is set by AndroidManifest.xml and doesn't change. Even if an app detects root, SuperSU or some similar su manager app will force the user to allow that app to access root, and even then, that app would have to be an Xposed module in order to be able to potentially overwrite the call that we use to handle package loading.

In short, it seems that something else might be causing your false positive.
